### PR TITLE
Subscribe to all Azure SDK diagnostic sources

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/7982

We've changed names of DiagnosticSources in Azure SDK, now every library gets it's own name to facilitate filtering in the future.

Change current implementation to subcribe based on the prefix.

cc @SergeyKanzhelev

### CODE REVIEW

File: src/OpenTelemetry/Collector/DiagnosticSourceSubscriber.cs [hunk 1]
```diff
@@ -71,8 +86,13 @@ public void Dispose()
...
             }
 
-            this.listenerSubscription?.Dispose();
-            this.listenerSubscription = null;
+            lock (this.listenerSubscriptions)
+            {
+                foreach (var listenerSubscription in this.listenerSubscriptions)
+                {
+                    listenerSubscription?.Dispose();
+                }
```
Comment (bruno-garcia) [inline src/OpenTelemetry/Collector/DiagnosticSourceSubscriber.cs]: Would it make sense to clear the list here to drop the references?
Something like:
```suggestion
                }
                this.listenerSubscriptions.Clear();
```

Comment (pakrym) [reply src/OpenTelemetry/Collector/DiagnosticSourceSubscriber.cs]: I don't think it really matters but I can do the change.

**Comment by pakrym** [comment]
I don't have merge permissions.

**Comment by SergeyKanzhelev** [comment]
I was waiting for a build to complete after merged with latest master =)

**Pull request merged**